### PR TITLE
fix getTransactions self tx duplication issue in xchain-eth

### PR DESCRIPTION
**Pull request by sarawut11**

### CODE REVIEW

File: packages/xchain-ethereum/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Fix
 
 - Fix `estimateGasLimit` to consider memo
+- Fix `getTransactions` self tx duplication issue
```
Comment (veado) [inline packages/xchain-ethereum/CHANGELOG.md:6]: It goes into next version 0.11.2. Please prepare it to bump xchain-eth after merging this PR.

File: packages/xchain-ethereum/src/etherscan-api.ts [hunk 2]
```diff
@@ -78,7 +78,14 @@ export const getETHTransactionHistory = async ({
...
 
   const ethTransactions: ETHTransactionInfo[] = await axios.get(url).then((response) => response.data.result)
-  return ethTransactions.filter((tx) => !bn(tx.value).isZero()).map(getTxFromEthTransaction)
+  const filterTxs = ethTransactions.filter((tx) => tx.from !== tx.to)
+  let selfTxs = ethTransactions.filter((tx) => tx.from === tx.to)
+  while (selfTxs.length) {
+    const selfTx = selfTxs[0]
+    filterTxs.push(selfTx)
+    selfTxs = selfTxs.filter((tx) => tx.hash !== selfTx.hash)
+  }
```
Comment (veado) [inline packages/xchain-ethereum/src/etherscan-api.ts:87]: Pls extract filter function + write a test. Use same filter function in `getTokenTransactionHistory` (as youi have already done it)

**Review by veado — CHANGES_REQUESTED** [review]

File: packages/xchain-ethereum/CHANGELOG.md (hunk 1 above)
Comment (sarawut11) [reply packages/xchain-ethereum/CHANGELOG.md:6]: 0.11.1 wasn't released yet, I forgot it to release last time so I wanted to involve this fix in 0.11.1 and release after that.

**Review by veado — APPROVED** [review]
LGTM

**Pull request merged**